subsys/nrf_security/src/drivers/nrf_oberon/Kconfig

```diff
@@ -1,353 +1,673 @@
+		   PSA_NEED_OBERON_TLS12_PSK_TO_MS          || \
+		   PSA_NEED_OBERON_PBKDF2_HMAC              || \
+		   PSA_NEED_OBERON_PBKDF2_AES_CMAC_PRF_128  || \
+		   PSA_NEED_OBERON_ECJPAKE_TO_PMS
+
+
+# Oberon PAKE Driver
 
-config PSA_CRYPTO_DRIVER_ECC_SECP_R1_256_OBERON
+config PSA_NEED_OBERON_ECJPAKE_P256
 	bool
-	prompt "PSA ECC secp256r1 - oberon" if !PSA_PROMPTLESS
-	default y if !PSA_DEFAULT_OFF && PSA_WANT_ECC_SECP_R1_256 && \
-		     !PSA_CRYPTO_DRIVER_ECC_SECP_R1_256_CC3XX
+	default y
+	depends on PSA_WANT_ALG_JPAKE
+	depends on (PSA_WANT_ALG_SHA_1   && !PSA_ACCEL_ECJPAKE_P256_SHA1)   || \
+		   (PSA_WANT_ALG_SHA_224 && !PSA_ACCEL_ECJPAKE_P256_SHA224) || \
+		   (PSA_WANT_ALG_SHA_256 && !PSA_ACCEL_ECJPAKE_P256_SHA256) || \
+		   (PSA_WANT_ALG_SHA_384 && !PSA_ACCEL_ECJPAKE_P256_SHA384) || \
+		   (PSA_WANT_ALG_SHA_512 && !PSA_ACCEL_ECJPAKE_P256_SHA512)
 
-config PSA_CRYPTO_DRIVER_ECC_SECP_R1_384_OBERON
+config PSA_NEED_OBERON_JPAKE_DRIVER
 	bool
-	prompt "PSA ECC secp384r1 - oberon" if !PSA_PROMPTLESS
-	default y if !PSA_DEFAULT_OFF && PSA_WANT_ECC_SECP_R1_384 && \
-		     !PSA_CRYPTO_DRIVER_ECC_SECP_R1_384_CC3XX
+	default y
+	depends on PSA_NEED_OBERON_ECJPAKE_P256
 
-config PSA_CRYPTO_DRIVER_ECC_TWISTED_EDWARDS_255_OBERON
+
+config PSA_NEED_OBERON_SPAKE2P_P256
 	bool
-	prompt "PSA ECC Curve Ed25519 support - oberon" if !PSA_PROMPTLESS
-	default y if !PSA_DEFAULT_OFF && PSA_WANT_ECC_TWISTED_EDWARDS_255
+	default y
+	depends on PSA_WANT_ALG_JPAKE
+	depends on (PSA_WANT_ALG_SHA_1   && !PSA_ACCEL_SPAKE2P_P256_SHA1)   || \
+		   (PSA_WANT_ALG_SHA_224 && !PSA_ACCEL_SPAKE2P_P256_SHA224) || \
+		   (PSA_WANT_ALG_SHA_256 && !PSA_ACCEL_SPAKE2P_P256_SHA256) || \
+		   (PSA_WANT_ALG_SHA_384 && !PSA_ACCEL_SPAKE2P_P256_SHA384) || \
+		   (PSA_WANT_ALG_SHA_512 && !PSA_ACCEL_SPAKE2P_P256_SHA512)
 
-config PSA_CRYPTO_DRIVER_RSA_KEY_SIZE_1024_OBERON
+config PSA_NEED_OBERON_SPAKE2P_DRIVER
 	bool
 	default y
-	depends on TFM_PSA_TEST_CRYPTO || \
-		   TFM_REGRESSION_S || \
-		   TFM_REGRESSION_NS
-	help
-	  Prompt-less configuration to allow RSA key size 1024 bits for testing
-	  RSA with 1024 bit keys is not recommended for new designs.
-	  Please see https://www.keylength.com/
+	depends on PSA_NEED_OBERON_SPAKE2P_P256
 
-config PSA_CRYPTO_DRIVER_ALG_JPAKE_OBERON
+
+config PSA_NEED_OBERON_SRP_6_3072
 	bool
-	prompt "PSA J-PAKE support - oberon" if !PSA_PROMPTLESS
-	default y if !PSA_DEFAULT_OFF && PSA_WANT_ALG_JPAKE
-	select EXPERIMENTAL
-	depends on PSA_CORE_OBERON
+	default y
+	depends on PSA_WANT_ALG_SRP_6
+	depends on (PSA_WANT_ALG_SHA_1   && !PSA_ACCEL_SRP_6_3072_SHA1)   || \
+		   (PSA_WANT_ALG_SHA_224 && !PSA_ACCEL_SRP_6_3072_SHA224) || \
+		   (PSA_WANT_ALG_SHA_256 && !PSA_ACCEL_SRP_6_3072_SHA256) || \
+		   (PSA_WANT_ALG_SHA_384 && !PSA_ACCEL_SRP_6_3072_SHA384) || \
+		   (PSA_WANT_ALG_SHA_512 && !PSA_ACCEL_SRP_6_3072_SHA512)
 
-config PSA_CRYPTO_DRIVER_ALG_SPAKE2P_OBERON
+config PSA_NEED_OBERON_SRP_DRIVER
```

Comment:
It would be nice if every WANT_ALG had a corresponding NEED option.

In that scheme this would be SRP_6.

When this is SRP, it is not clear to me if there is a system here at all.

Reply:
So this comes from Oberon so I will not update it here.

But I already gave feedback about this because I don't want the symbols with the _DRIVER suffix to be algorithm specific.